pkg/provider/ibm/provider.go

```diff
@@ -345,6 +410,25 @@ func (ibm *providerIBM) GetSecretMap(ctx context.Context, ref esv1beta1.External
 
 		return secretMap, nil
 
+	case sm.CreateSecretOptionsSecretTypeKvConst:
+		secret, err := getSecretByType(ibm, &secretName, sm.CreateSecretOptionsSecretTypeKvConst)
+		if err != nil {
+			return nil, err
+		}
+
+		secretData := secret.SecretData.(map[string]interface{})
+		secretPayload := secretData["payload"].(string)
```

Comment:
This will crash when running GetSecretMap. The contents of secretData is not going to be a string, but rather an interface.